bot/cogs/bot.py

```diff
@@ -60,6 +62,23 @@ def __init__(self, bot: AutoShardedBot):
 
         await ctx.invoke(self.info)
 
+    async def on_message(self, msg: Message):
+        if msg.content.count("\n") >= 3:
+            try:
+                tree = ast.parse(msg.content)
+                # Attempts to parse the message into an AST node.
+                # Invalid Python code will raise a SyntaxError.
+                if not all(isinstance(node, ast.Expr) for node in tree.body):
+                    # we dont want multiple lines of single words,
+                    # they would be syntactically valid Python but could also be
+                    # just some random multiline text someone is sending.
+                    formatted = f"```python\n{msg.content}\n```"
+                    if len(formatted) <= 2000:
```

Comment:
Not going to start this as a requested change but do we want the bot to replace the users content, or do we want the bot to send a message informing the user of how to code block?

Reply:
i believe you cannot edit other peoples msgs on discord can you?

Reply:
I would say replace because some user will take some time to actually understand how to to use codeblock

Reply:
we want to inform the user. The goal is for the user to learn how to do this in the future.

Reply:
They seem to grasp our current `py!tag codeblock` text soon enough, and I think it's better letting them know for future.

Reply:
I actually think this is a critical point. replacing their message teaches them nothing. they'll go "oh cool" and have zero motivation to learn how to do it.

Reply:
Text from current tag:

If you are going to post code in Discord, please use syntax highlighted blocks, as it makes it more legible for other users.

To do this, you should input your content like this:

\\\`\\\`\\\`python
print("Hello world!")
\\\`\\\`\\\`

This will result in the following:
\`\`\`
print("Hello world!")
\`\`\`